SampleGallery/Samples/SDK 17134/RadialGradients/RadialGradients.xaml.cs

```diff
@@ -0,0 +1,310 @@
+                _changeButtonGradientStop2.InsertKeyFrame(1, _orangeRed);
+                _changeButtonGradientStop2.Duration = TimeSpan.FromSeconds(2);
+
+                _BBGradientStop2.StartAnimation(nameof(_BBGradientStop2.Color), _changeButtonGradientStop2);
+
+                // Creating the animation for outer visual to create the pulsing effect that radiates from the center of the button out
+                // Animation for the first stop of the radial gradient brush applied on the pulsing sprite visual
+                _stop1OffsetAnim = _compositor.CreateScalarKeyFrameAnimation();
+                _stop1OffsetAnim.InsertKeyFrame(0, 0);
+                _stop1OffsetAnim.InsertKeyFrame(1f, 1f);
+                _stop1OffsetAnim.Duration = TimeSpan.FromSeconds(1);
+                _stop1OffsetAnim.IterationCount = 50;
+
+                _PBGradientStop1.StartAnimation("Offset", _stop1OffsetAnim);
+
+                // Animation for the second stop of the radial gradient brush applied on the pulsing sprite visual
+                _stop2offsetAnim2 = _compositor.CreateScalarKeyFrameAnimation();
+                _stop2offsetAnim2.InsertKeyFrame(0, 0);
+                _stop2offsetAnim2.InsertKeyFrame(1f, 1f);
+                _stop2offsetAnim2.Duration = TimeSpan.FromSeconds(1);
+                _stop2offsetAnim2.IterationCount = 50;
+                _stop2offsetAnim2.DelayTime = TimeSpan.FromSeconds(.25f);
+
+                _PBGradientStop2.StartAnimation("Offset", _stop2offsetAnim2);
+
+                // When the stopwatch starts, the color for the gradient stops are changed to Alice Blue so that the pulse is visible
+                _pulseColor = _compositor.CreateColorKeyFrameAnimation();
+                _pulseColor.InsertKeyFrame(0, _transparent);
+                _pulseColor.InsertKeyFrame(.99f, _aliceBlue);
+                _pulseColor.InsertKeyFrame(1, _transparent);
+                _pulseColor.Duration = TimeSpan.FromSeconds(1);
+                _pulseColor.IterationBehavior = AnimationIterationBehavior.Forever;
+
+                _PBGradientStop1.StartAnimation(nameof(_PBGradientStop1.Color), _pulseColor);
+
+                // Create the animation that animates the scale of the pulsing sprite visual
+                _scale = _compositor.CreateVector3KeyFrameAnimation();
+                _scale.InsertKeyFrame(0, new Vector3(0, 0, 0));
+                _scale.InsertKeyFrame(1, new Vector3(1, 1, 0));
+                _scale.Duration = TimeSpan.FromSeconds(1);
+                _scale.IterationCount = 50;
+
+                _pulseVisual.StartAnimation("Scale", _scale);
+
+                // Set up the stopwatch to reset each it is started again
+                _dt = new DispatcherTimer();
+                _stopwatch = new Stopwatch();
+                _dt.Interval = TimeSpan.FromMilliseconds(0.00f);
+                _dt.Tick += Dt_Tick;
+                _dt.Start();
+                _stopwatch.Start();
```

Comment:
Seeing as you're starting it straight after creating it, just use Stopwatch.StartNew()

Reply:
Instead of stopwatch.Start()?

Reply:
Yep. It does the work of 2 lines in 1.

Reply:
updated this